packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1068,6 +1070,31 @@ export class ContainerRuntime extends EventEmitter
         return { snapshot, state };
     }
 
+    public async raceToSummarize(): Promise<void> {
+        this.verifyNotClosed();
+
+        // Stop ops from being processed by runtime
+        this.racingToSummarize = true;
+
+        const sequenceNumber = this.deltaManager.lastSequenceNumber;
+        const summaryCollection = this.summarizer.summaryCollection;
+
+        // Check for early completion
+        const latestAckSeq = summaryCollection.latestAck?.summaryOp.referenceSequenceNumber;
+        if (latestAckSeq && latestAckSeq >= sequenceNumber) {
+            return;
+        }
+
+        // Try to summarize only once and fail otherwise. In the future, may want
+        // to retry with back-off until success is reached.
+        this.generateSummary(false, false, true).catch((error) => {
```

Comment:
can you please add comments for each argument?
I.e. 
```
this.generateSummary(false /* arg1*/ , false /* arg2*/, true /* arg3*/)
```
or
```
this.generateSummary(
    false, // arg1
    false, // arg2
    true); // arg3
```

Reply:
Addressed by using object parameter